### PR TITLE
Port DmIcon & Image components from TG

### PR DESCRIPTION
**Pull request by AyIong**
## What Does This PR Do
Ports DmIcon from https://github.com/tgstation/tgstation/pull/82533 and early_assets subsystem for it
And Image components from TG (Didn't find PR)

Replacing base64 sprites generated during init with DmIcon in Stack Craft (GrayScale sprites still generates, but it's okay, instead of 300+ icon2base64 calls in Stack Craft, it comes out to about 14)
Replacing Assets with DmIcon in Prize Counter

Added DmIcon support to ImageButton and replaced  with Image component

## Why It's Good For The Game
The most modern, convenient and fastest way to add any image to the TGUI
So how do I add an image? VERY easy! You just need to pass icon and icon_state of the object to TGUI, and use them in DmIcon or ImageButton 
No need to generate images anymore (Unless they are grayscale or built from overlays).

## Images of changes
Nothing to show, result the same, maybe UI's shown a little faster, cause it doesn't need to wait images, they load async
But, can show icon2base64 overtime before and after

| Before | After |
| - | - |
| image | image |

## Testing
Very many times opened Ui's using DmIcon and normal Image with Assets (Before removing Assets from Prize Counter)
Also tested on Byond 516 with CPU slowdown 20x and in SmartFridge with test UI (Reverted but video bellow)
<details>
<summary> Smartfridge test UI </summary>

https://github.com/user-attachments/assets/eb35d2f8-a857-4bbf-8335-6d20c4a5c879

</details>

I fixed all the problems I noticed, but maybe I missed some small things as usual.

<hr>

### Declaration

- [x] I confirm that I either do not require [pre-approval](https://github.com/ParadiseSS13/Paradise/blob/master/CODE_OF_CONDUCT.md#types-of-changes-that-need-approval) for this PR, or I have obtained such approval and have included a screenshot to demonstrate this below.

<hr>

## Changelog
NPFC

### CODE REVIEW

File: code/game/objects/items/stacks/stack.dm [hunk 1]
```diff
@@ -250,12 +250,18 @@
...
 
 	data["uid"] = recipe.UID()
 	data["required_amount"] = recipe.req_amount
 	data["result_amount"] = recipe.res_amount
 	data["max_result_amount"] = recipe.max_res_amount
-	data["image"] = recipe.image
+	data["icon"] = result.icon
+	data["icon_state"] = result.icon_state
+
+	// DmIcon cannot paint images. So, if we have grayscale sprite, we need ready base64 image.
```
Comment (lewcc) [inline code/game/objects/items/stacks/stack.dm:262]: Is this in reference to GAGS?

File: code/modules/asset_cache/asset_list.dm [hunk 2]
```diff
@@ -5,18 +5,31 @@
 GLOBAL_LIST_EMPTY(asset_datums)
 
 //get an assetdatum or make a new one
-/proc/get_asset_datum(type)
+//does NOT ensure it's filled, if you want that use get_asset_datum()
+/proc/load_asset_datum(type)
 	return GLOB.asset_datums[type] || new type()
 
+/proc/get_asset_datum(type)
+	var/datum/asset/loaded_asset = GLOB.asset_datums[type] || new type()
```
Comment (lewcc) [inline code/modules/asset_cache/asset_list.dm:13]: Is it worth saving this new value to cache if it doesn't exist?

**Review by lewcc — COMMENTED** [review]
I don't know too much about what's going on here tbh, just some questions

File: code/modules/asset_cache/asset_list.dm (hunk 2 above)
Comment (AyIong) [reply code/modules/asset_cache/asset_list.dm:13]: Not sure, just copied that from TG.
It just works™️

Works pretty good btw

File: code/game/objects/items/stacks/stack.dm (hunk 1 above)
Comment (AyIong) [reply code/game/objects/items/stacks/stack.dm:262]: Nah, TG haven't images on Stack Crafting UI, but we need to paint ~14 grayscale sprites for it
Anyway that's better than make all 300+ base64 sprites

**Review by Burzah — APPROVED** [review]
I am willing to give this tentative approval. It seems to follow the same lines as the original implementation on tg. I would like another reviewer to look at it, as this creates a new subsystem. Changes to the existing UI that this PR will affect look good, as well as the existing components.

**Pull request merged**